### PR TITLE
fix: integrate call spread contract with Oracle API.

### PR DESCRIPTION
**Pull request by Chris-Hibbert**

### CODE REVIEW

File: packages/zoe/src/contracts/callSpread.js [hunk 1]
```diff
@@ -133,17 +135,24 @@ const start = zcf => {
...
-      .priceAtTime(terms.expiration, terms.underlyingAmount)
-      .then(price => payoffOptions(price));
+    E(terms.priceAuthority)
+      .priceAtTime(
+        terms.timer,
+        terms.expiration,
+        terms.underlyingAmount,
+        strikeBrand,
+      )
+      .then(quoteIssuer.getAmountOf)
```
Comment (michaelfig) [inline packages/zoe/src/contracts/callSpread.js:154]:
```suggestion
    E(quoteIssuer).getAmountOf(E(terms.priceAuthority)
      .priceAtTime(
        terms.timer,
        terms.expiration,
        terms.underlyingAmount,
        strikeBrand,
      ))
```

**Review by michaelfig — COMMENTED** [review]
May need some more API updates.  I'm not sure.

File: packages/zoe/src/contracts/callSpread.js [hunk 2]
```diff
@@ -71,7 +71,9 @@ const start = zcf => {
 
   const terms = zcf.getTerms();
   const {
-    maths: { Collateral: collateralMath, Strike: strikeMath },
+    maths: { Collateral: collateralMath, Strike: strikeMath, Quote: quoteMath },
+    brands: { Strike: strikeBrand },
+    issuers: { Quote: quoteIssuer },
```
Comment (katelynsills) [inline packages/zoe/src/contracts/callSpread.js:76]: Does this contract need to use the quoteIssuer?

File: packages/zoe/src/contracts/callSpread.js [hunk 3]
```diff
@@ -133,17 +135,24 @@ const start = zcf => {
     return floorDivide(multiply(PERCENT_BASE, numerator), denominator);
   }
 
-  function payoffOptions(price) {
-    // either offer might be exercised late, so we pay the two seats separately.
+  function payoffOptions(priceQuoteAmount) {
+    const { Price: price } = quoteMath.getValue(priceQuoteAmount)[0];
```
Comment (katelynsills) [inline packages/zoe/src/contracts/callSpread.js:139]: I think `price` is already lower case

File: packages/zoe/src/contracts/callSpread.js (hunk 1 above)
Comment (katelynsills) [reply packages/zoe/src/contracts/callSpread.js:154]: Can we use the `quoteAmount` part of the `{ quoteAmount, quotePayment }` that we discussed this morning so we don't have to do `getAmountOf`?

Comment (katelynsills) [reply packages/zoe/src/contracts/callSpread.js:154]: And also then we don't need to know about `quoteIssuer` so we can drop that

File: packages/zoe/test/unitTests/contracts/test-callSpread.js [hunk 4]
```diff
@@ -32,25 +39,40 @@ function makeFakePriceAuthority(
...
+                Price: strikePrice,
+                timer,
+                timestamp: now,
+              },
+            ]),
+          ),
+        );
+      });
+  }
+
```
Comment (katelynsills) [inline packages/zoe/test/unitTests/contracts/test-callSpread.js:63]: Why define `getRecentPrice` when you only need `priceAtTime` to test?

**Review by katelynsills — CHANGES_REQUESTED** [review]
I think we should remove the quoteIssuer and rely on the `quoteAmount` only in this contract.

**Comment by Chris-Hibbert** [comment]
I'm going to merge this into the base PR and address comments once rather than try to keep 2 PRs synced.

File: packages/zoe/src/contracts/callSpread.js (hunk 3 above)
Comment (Chris-Hibbert) [reply packages/zoe/src/contracts/callSpread.js:139]: yup.

**Comment by katelynsills** [comment]
> I'm going to merge this into the base PR and address comments once rather than try to keep 2 PRs synced.

sounds good!

File: packages/zoe/test/unitTests/contracts/test-callSpread.js (hunk 4 above)
Comment (Chris-Hibbert) [reply packages/zoe/test/unitTests/contracts/test-callSpread.js:63]: I simplified the helper function.

File: packages/zoe/src/contracts/callSpread.js (hunk 2 above)
Comment (Chris-Hibbert) [reply packages/zoe/src/contracts/callSpread.js:76]: Not any more.

**Review by Chris-Hibbert — COMMENTED** [review]
changes made in base PR after merge. Thanks for the suggestions.

**Pull request merged**